src/pybind/mgr/orchestrator/module.py

```diff
@@ -1256,6 +1260,40 @@ def _apply_snmp_gateway(self,
 
         return self._apply_misc([spec], dry_run, format, no_overwrite)
 
+    @_cli_write_command('orch apply jaeger')
+    def _apply_jaeger(self,
+                      es_nodes: Optional[str] = None,
+                      placement: Optional[str] = None,
+                      unmanaged: bool = False,
+                      dry_run: bool = False,
+                      format: Format = Format.plain,
+                      no_overwrite: bool = False,
+                      inbuf: Optional[str] = None) -> HandleCommandResult:
+        """Apply jaeger tracing services"""
+        if inbuf:
+            raise OrchestratorValidationError('unrecognized command -i; -h or --help for usage')
```

Comment:
@adk3798 
is there a yaml format that I should follow? like the Service spec one
or the idea is just to pass the `apply_jaeger` arguments?

Reply:
You already have the `TracingSpec` created which is a lot of the work. The only complicated bit here is because this apply command creates multiple services (it's the first example of this in cephadm) you will need special handling around https://github.com/ceph/ceph/blob/master/src/pybind/mgr/orchestrator/module.py#L1054-L1072. I'd recommend allowing users to apply a spec with a service type of `jaeger` or `tracing` or something like that (whatever you choose would have to be added to the list of acceptable service type for `TracingSpec`) and then instead of just appending the spec to the list as is done at the end of the block now, if the service type matches what you choose have it create the 3 or 4 specs you need and append all of them. You'd probably just want a separate function that takes the `TracingSpec` and returns the 3 or 4 specs for each service sort of like how the end bits of `_apply_jaeger` works. You could then call to that and append all the specs returned in the end of the code block.

Reply:
thanks
added in [ea059b0](https://github.com/ceph/ceph/pull/44493/commits/ea059b042d815d5d8cde0475feba03f5336b3da1)